viewer/src/tools/HtmlOverlayTool.ts

```diff
@@ -0,0 +1,216 @@
+ * useful to create HTML overlays to highlight information about key positions in the 3D model.
+ *
+ * Attached elements *must* have CSS style 'position: absolute'. It's also recommended
+ * in most cases to have styles 'pointerEvents: none' and 'touchAction: none' to avoid
+ * interfering with 3D navigation. Consider also applying 'transform: translate(-50%, -50%)'
+ * to anchor the center of the element rather than the top-left corner. In some cases the
+ * `zIndex`-attribute is necessary for the element to appear on top of the viewer.
+ *
+ * @example
+ * ```js
+ * const el = document.createElement('div');
+ * el.style.position = 'absolute'; // Required!
+ * // Anchor to center of element
+ * el.style.transform = 'translate(-50%, -50%)';
+ * // Avoid being target for events
+ * el.style.pointerEvents = 'none;
+ * el.style.touchAction = 'none';
+ * // Render in front of other elements
+ * el.style.zIndex = 10;
+ *
+ * el.style.color = 'red';
+ * el.innerHtml = '<h1>Overlay</h1>';
+ *
+ * const overlayTool = new HtmlOverlayTool(viewer);
+ * overlayTool.add(el, new THREE.Vector3(10, 10, 10));
+ * // ...
+ * overlayTool.remove(el);
+ * // or, to remove all attached elements
+ * overlayTool.clear();
+ 
+ * // detach the tool from the viewer
+ * overlayTool.dispose();
+ * ```
+ */
+export class HtmlOverlayTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _htmlOverlays: Map<HTMLElement, HtmlOverlayElement> = new Map();
+
+  private readonly _onSceneRenderedHandler: SceneRenderedDelegate;
+  // Allocate variables needed for processing once to avoid allocations
+  private readonly _preallocatedVariables = {
+    camPos: new THREE.Vector3(),
+    camNormal: new THREE.Vector3(),
+    point: new THREE.Vector3(),
+    nearPlane: new THREE.Plane(),
+    farPlane: new THREE.Plane(),
+    position2D: new THREE.Vector2()
+  };
+
+  private get viewerDomElement(): HTMLElement {
+    return this._viewer.domElement;
+  }
+
+  private get viewerCamera(): THREE.PerspectiveCamera {
+    return this._viewer.getCamera();
+  }
+
+  private get viewerRenderer(): THREE.WebGLRenderer {
+    return this._viewer.renderer;
+  }
+
+  constructor(viewer: Cognite3DViewer) {
+    super();
+
+    this._onSceneRenderedHandler = this.onSceneRendered.bind(this);
+    this._viewer = viewer;
+    this._viewer.on('sceneRendered', this._onSceneRenderedHandler);
+  }
+
+  /**
+   * Removes all elements and detaches from the viewer.
+   * @override
+   */
+  dispose(): void {
+    this._viewer.off('sceneRendered', this._onSceneRenderedHandler);
+    this.clear();
+    super.dispose();
+  }
+
+  /**
+   * Registers a HTML overlay that will be updated on rendering.
+   *
+   * @param htmlElement
+   * @param position3D
+   * @param options
+   */
+  add(htmlElement: HTMLElement, position3D: THREE.Vector3, options: HtmlOverlayOptions = {}) {
+    this.ensureNotDisposed();
+    if (htmlElement.style.position !== 'absolute') {
```

Comment:
It will give a false alarm in case if you have some HTML element styled with css (not style property). In that case, you need computed styles.

```css
.label { 
  position: absolute;
  /* other styles */
}
```

```js
createOverlayHtmlElement() {
  let el = document.createElement('div');
  el.classList.add('label');
  return el
}

Reply:
Oh, thx